osu.Game/Rulesets/Edit/HitObjectComposer.cs

```diff
@@ -56,6 +58,26 @@ public abstract class HitObjectComposer<TObject> : HitObjectComposer, IPlacement
         [Resolved]
         protected IBeatSnapProvider BeatSnapProvider { get; private set; }
 
+        /// <summary>
+        /// Whether this composer supports a "distance spacing" multiplier for distance snap grids.
+        /// </summary>
+        /// <remarks>
+        /// Setting this to <see langword="true"/> displays a "distance spacing" slider and allows this composer to configure the value of <see cref="BeatmapInfo.DistanceSpacing"/>.
+        /// </remarks>
+        protected abstract bool SupportsDistanceSpacing { get; }
```

Comment:
Was there a reason for doing it this way, rather than having it a per-ruleset implementation?

Reply:
I can see it being shared across osu! and osu!catch, but I think I could move out the whole distance spacing handler to its own component and use it in each ruleset implementation.

Reply:
Not 100% sure yet. Let me dwell on this a bit longer.

Reply:
I think I would personally slightly prefer not having this here and this being attached in rulesets in a self-contained way if possible. I'm worried that it will become a precedent for all sorts of abstract flags and switches on `HitObjectComposer` and I'm not at all sure that it's a path we want to be taking.